src/main/java/com/microsoft/spring/data/gremlin/telemetry/TelemetryEventData.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See LICENSE in the project root for
+ * license information.
+ */
+package com.microsoft.spring.data.gremlin.telemetry;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.AccessLevel;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.Setter;
+import org.springframework.util.Assert;
+
+import java.time.Instant;
+import java.util.Map;
+
+@Getter
+public class TelemetryEventData {
+
+    private final String name;
+
+    @JsonProperty("iKey")
+    private final String instrumentationKey;
+
+    private final Tags tags = new Tags("Spring-on-azure", "Java-maven-plugin");
+
+    private final EventData data = new EventData("EventData");
+
+    private final String time;
+
+    public TelemetryEventData(String eventName, @NonNull Map<String, String> properties) {
+        Assert.hasText(eventName, "Event name should contain text.");
+
+        name = "Microsoft.ApplicationInsights.Event";
```

Comment:
The name is initialized once, move initialization with the declaration? And `name` seems not used?

Reply:
make sense, will change to trace for less misleading to users.